web/client-api/src/main/java/io/deephaven/web/client/api/subscription/TableViewportSubscription.java

```diff
@@ -250,6 +250,9 @@ public void setInternalViewport(double firstRow, double lastRow, Column[] column
             Boolean isReverseViewport) {
         // Until we've created the stream, we just cache the requested viewport
         if (status == Status.STARTING) {
+            if (firstRow < 0 || firstRow > lastRow) {
+                throw new IllegalArgumentException("Invalid viewport row range: " + firstRow + " to " + lastRow);
+            }
```

Comment:
Guessing you added this because it was in DHE?

Reply:
Correct - trying to slightly better match semantics, especially since we removed the check in RangeSet that the ranges are ordered properly (as it was causing performance issues on subscriptions to large tables).

This is one of the few cases where we actually throw when calling setViewport in both APIs. When we get an api design for subscriptions that also handles preview and reverse viewports, we'll be more deliberate about how to handle these classes of errors...